File: LongestRepeatingCharacterReplacement.js
var characterReplacement = function(s, k) {
    var maxFrequency = 0;
    var result = 0;
    const count = new Map();
    var left = 0;
    for (let i = 0; i < s.length; i++) {
        if (count.has(s.charAt(i))) {
            count.set(s.charAt(i), count.get(s.charAt(i)) + 1);
        } else {
            count.set(s.charAt(i), 1);
        }

        if (count.get(s.charAt(i)) > maxFrequency) {
            maxFrequency = count.get(s.charAt(i));
        }

        while ((i + 1 - left) - maxFrequency > k) {
            count.set(s.charAt(left), count.get(s.charAt(left)) - 1);
            left++;
        } 
        result = Math.max(result, i + 1 - left);
    }
    return result;
};

File: LongestRepeatingCharacterReplacement.java
import java.util.*;

public class LongestRepeatingCharacterReplacement {
    public int characterReplacement(String s, int k) {
        HashMap<Character, Integer> hm = new HashMap<>();
        int maxF = 0;
        int left = 0;
        int result = 0;

        for (int i = 0; i < s.length(); i++) {
            if (hm.containsKey(s.charAt(i))) {
                hm.put(s.charAt(i), hm.get(s.charAt(i)) + 1);
            } else {
                hm.put(s.charAt(i), 1);
            }
            maxF = Math.max(maxF, hm.get(s.charAt(i)));

            while (i - left + 1 - maxF > k) {
                hm.put(s.charAt(left), hm.get(s.charAt(left)) - 1);
                left++;
            }

            result = Math.max(i - left + 1, result);
        }

        return result;
    }
}
